lib/grp.gi

```diff
@@ -528,13 +532,22 @@ InstallMethod( IsSolvableGroup,
     "generic method for groups",
     [ IsGroup ],
     function ( G )
-    local   S;          # derived series of <G>
+    local   S,          # derived series of <G>
+            isAbelian,  # true if <G> is abelian
+            isSolvable; # true if <G> is solvable
 
     # compute the derived series of <G>
     S := DerivedSeriesOfGroup( G );
 
     # the group is solvable if the derived series reaches the trivial group
-    return IsTrivial( S[ Length( S ) ] );
+    isSolvable := IsTrivial( S[ Length( S ) ] );
+
+    # set IsAbelian filter
+    isAbelian := isSolvable and Length( S ) <= 2;
+    Assert(2, IsAbelian(G) = isAbelian);
```

Comment:
What is the point of this assert? It adds several seconds to your tests, as tests are run at assertion level 2 right now.

I'd either remove it, or increase the level to 3.